crates/turborepo-lib/src/run/summary/spaces.rs

```diff
@@ -257,8 +360,14 @@ mod tests {
             "rauchg".to_string(),
         ))?;
 
+        let mut join_set = tokio::task::JoinSet::new();
         for task_summary in tasks {
-            spaces_client_handle.finish_task(task_summary).await?;
+            let task_client = spaces_client_handle.task_client();
+            join_set.spawn(async move { task_client.send_task(task_summary).await });
```

Comment:
Slightly more realistic test where we finish tasks concurrently instead of serially.

Reply:
nit: why not pass the future directly?
```suggestion
            let task_client = spaces_client_handle.task_client();
            join_set.spawn(task_client.send_task(task_summary));
```

Reply:
The future needs ownership of the client which AFAIK `async move` is the best way to ensure the resulting future has ownership. I could do something with `move` closure, but at that point it's less readable IMO than an `async move` block.